### DISCUSSION TITLE
Proposal to change some defaults

### DISCUSSION THREAD

**Discussion by raphaelm** [pretix/pretix#2146] [Ideas]
pretix has thousands¹ of settings and we often tell users "don't worry, only look for what you need, everything has a sane default" and generally that is true. However, some of the defaults are historic – when we introduced new options we set the default to the previous behavior. I think the time has come to change some defaults based on the usage patterns we see.

On a technical level, we wouldn't really change the default in the settings mechanism, but instead just set them on new events. (One exception might be the design change.) So the change would not affect existing or copied events at all and would only be relevant if new events are created from scratch. We'd probably need a similar mechanism for organizers.

### Shop design

In #1585, we introduced more shop design settings, but didn't change the defaults. So a newly created event still looks like it did 6 years ago:

![Bildschirmfoto am 2021-07-09 um 13 40 04](https://user-images.githubusercontent.com/64280/125072811-5a189a00-e0bb-11eb-85ea-fb013700df73.png)

Looking at customer shops, I find all those with a page background set, even if just to a light grey, much more pleasant to look at. Also, I think the header with language selection, login and back-link is much easier to graph visually.

![Bildschirmfoto am 2021-07-09 um 13 41 05](https://user-images.githubusercontent.com/64280/125072852-6ac91000-e0bb-11eb-9dcc-7c2d93e4da8d.png)

So I propose making this the default for newly created shops, with something like #f5f5f5 as the background (or even lighter). Happy especially for input from @wiffbi here.

### Name format

Historically, when pretix asked for a "name" e.g. in invoice addresses or attendee profiles, we only provided a single text field. The reason for this was internationalization. [Not everywhere in the world people have a first name and a last name](https://www.kalzumeus.com/2010/06/17/falsehoods-programmers-believe-about-names/). After long discussion (read #978 if you're interested), we introduced a flexible concept of "name formats", allowing to break name fields differently depending on culture.

This setting is easily overlooked when setting up pretix, and once you have a few hundred names in there and want to sort your attendee list by last name, it's too late to be educated about that settings. Since 99% of known pretix applications are in cultures where at least first name / last name is a thing, I propose making the `given_family` format the default.

If someone wants salutations, titles, middle names, or just one field, they can always easily go back, but `given_family` is a better place to start than `full_name`.

If we see lots of pretix usage in other cultures in the future, we can still make the default e.g. dependent on the region setting of the organizer.

### Attach invoices to emails

No idea why that isn't the default yet.

### Shop redirect

So this is a tough one. The option "Directly redirect to check-out after a product has been added to the cart." causes the following change in behavior: After you selected a product, the system won't show you your cart and the list of products again, but instead directly send you to checkout.

Pro: The process is much quicker and more streamlined. People who just want a specific ticket do not even need to know there is such a thing as a cart. Probably great for conversion.

Con: You do not immediately see a confirmation of what you selected (could be fixed by expanding the cart on the first page of checkout), and you are not encouraged or given the option to add *more* stuff to your cart, which is especially relevant for event series where you otherwise might not figure out how to add tickets for other dates to the same cart.

Possible idea: Keep it as-is for event series and change it for non-series events? Or is that too confusing and we should just leave it as-is completely?

### Payment deadline

pretix currently defaults to a payment deadline of 14 days. A large portion of our users only offer instant-ish payment options such as credit card payments, in which case 1 day would be more appropriate. However, 14 days is a good default for every event that accepts payments by wire transfer.

I'm really torn on this one. An option would be changing the default to 1 day and giving out a fat warning when enabling bank transfers. Or the other way round? Any ideas?

Before anyone asks, [here's why we don't do deadlines per payment method](https://docs.pretix.eu/en/latest/user/faq.html#can-i-have-different-payment-deadlines-for-different-payment-methods).

### More?

Do you disagree? Do you see more settings we should change the default for? Now is the time!

### Footnotes

1: Yup. I ran ``SELECT count(*) from (select key from  pretixbase_event_settingsstore where key NOT LIKE 'ticketoutput_pdf_%' group by key) a;`` on pretix Hosted production, and it found 1164 distinct keys in the settings data store.

**Comment by wiffbi** [comment]
Regarding the background color, I think this is a good change. I helps to separate the secondary top-nav from content. The content itself probably would not necessarily need it, but it helps to. It might be worth a try to add the background to the footer-nav as well, so it is consistent with the top-nav. Generally a bit more padding might be helpful – but that has nothing to do regarding default settings. padding, white-space and vertical rhyhtm is a not a default settings change, but worth its own PR regarding UI optimizations.

Regarding „Shop redirect“: I think for single events this is great! On a single event page you have all products listed and just add everything you want and start checkout. You could argue that this behaviour is more like a dedicated „basket-page“ which directly starts you into the checkout process. The important part is, that you need to show the basket in the first step of checkout – either generally or only when a „add-to-basket-operation“ happened. Regarding a event-series, I think the current behaviour is okay. Or use the instant-checkout, but add the button „add tickets for a different date“ underneath the opened basket-view on the first checkout page.

The other changes also make sense IMHO.
